### PR TITLE
[NO QA]Add GitHub Action linting

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details
We have had a few failures and security vulnerabilities reported by this static analyzer, so I've done two things:
1. Fix all the issues (most were word globbing issues with `" "` needing to be applied)
2. Run the linter every commit along with `validateActionsAndWorkflows.sh`

### Fixed Issues

$ https://github.com/Expensify/App/issues/7703

### Tests
1. Merge this PR
2. Run a full deploy and verify none of the changes have broken anything 😬

### CODE REVIEW

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: .github/workflows/deployBlocker.yml [hunk 1]
```diff
@@ -18,10 +18,10 @@ jobs:
 
       - name: Get URL, title, & number of new deploy blocker (issue)
         if: ${{ github.event_name == 'issues' }}
+        env:
+          TITLE: ${{ github.event.issue.title }}
         run: |
-          echo "DEPLOY_BLOCKER_URL=${{ github.event.issue.html_url }}" >> $GITHUB_ENV
-          echo "DEPLOY_BLOCKER_NUMBER=${{ github.event.issue.number }}" >> $GITHUB_ENV
-          echo "DEPLOY_BLOCKER_TITLE=$(sed -e "s/'/'\\\\''/g; s/\`/\\\\\`/g; 1s/^/'/; \$s/\$/'/" <<< "'${{ github.event.issue.title }}'")" >> $GITHUB_ENV
+          { echo "DEPLOY_BLOCKER_URL=${{ github.event.issue.html_url }}"; echo "DEPLOY_BLOCKER_NUMBER=${{ github.event.issue.number }}"; echo "DEPLOY_BLOCKER_TITLE=$(sed -e "s/'/'\\\\''/g; s/\`/\\\\\`/g; 1s/^/'/; \$s/\$/'/" <<< "$TITLE")";} >> "$GITHUB_ENV"
```
[the remark this answers is not in the document]
Comment (AndrewGable) [reply .github/workflows/deployBlocker.yml:24]: It was complaining that I needed to lump the commands together, but I will try new lines with `\`

[the remark this answers is not in the document]
Comment (roryabraham) [reply .github/workflows/deployBlocker.yml:24]: Well ... still NAB because you didn't introduce this.

**Review by neil-marcellini — COMMENTED** [review]
Looks pretty good to me as much as I understand it.

**Comment by AndrewGable** [comment]
Thanks for the review @roryabraham and @neil-marcellini! I will pick up the review comments later today.

**Review by roryabraham — APPROVED** [review]

**Pull request merged**